src/level_set.cpp

```diff
@@ -139,45 +142,52 @@ void LevelSet::TemporalClearAndGenerateCostmap(                               \
     std::vector<frustum_model>::iterator frustum_it = frustums.begin();
     bool frustum_cycle = false;
 
+    const double time_since_marking = cur_time - cit_grid.getValue();
+
     for(frustum_it; frustum_it != frustums.end(); ++frustum_it)
     {
       if ( frustum_it->frustum.IsInside(this->IndexToWorld(pt_index)) )
       {
         frustum_cycle = true;
-        const double accel_decay_time = \
-                            GetAcceleratedDecayTime(frustum_it->accel_factor);
-        if (true) //cit_grid.getValue() < accel_decay_time)
+
+        // accel_decay_shift is the amount of time from the current time until
+        // the time a given mark is supposed to be cleared
+        const double accel_decay_shift = \
+          GetAcceleratedDecayShift(time_since_marking, \
+                                   frustum_it->accel_factor);
+        if (accel_decay_shift <= 0)
         {
-          // accelerate the values stored. Ticket #23 TODO
-          // if(!this->MarkLevelSetPoint(pt_index, \
-          //    cit_grid.getValue()-accel_decay_time))
-          // {
-          //   std::cout << "Failed to clear point." << std::endl;
-          // }
-          this->ClearLevelSetPoint(pt_index); // temp for testing
+          // expired by acceleration
+          if(!this->ClearLevelSetPoint(pt_index))
+          {
+            std::cout << "Failed to clear point." << std::endl;
+          }
         }
         else
         {
-          // expired by acceleration
-          // if(!this->ClearLevelSetPoint(pt_index))
-          // {
-          //   std::cout << "Failed to clear point." << std::endl;
-          // }
+          const double updated_mark = cur_time - accel_decay_shift;
```

Comment:
does this make sense for linear and exponential? (I don't know, not a loaded question)